1.8/administration/access-node/tunnel.md

```diff
@@ -0,0 +1,113 @@
+
+# VPN
+dcos tunnel provides you with full access to the DNS, masters, and agents from within the cluster. OpenVPN requires root privileges to configure these routes.
+
+# Using dcos tunnel
+
+## Prerequisites
+* DC/OS 1.8+
+* Docker
+* Requires [SSH access](/1.8/administration/access-node/sshcluster/) (key authentication only).
+* For VPN functionality, you must have [the OpenVPN client](https://openvpn.net/index.php/open-source/downloads.html) installed.
+* The dcos tunnel package must be installed: run `dcos package install tunnel-cli --cli` from the DC/OS CLI.
+
+**Note:** * Only Linux and OS X are supported currently.
+
+##  SOCKS
+Run the following command from the DC/OS CLI:
+
+```
+$ dcos tunnel socks
+```
+
+Configure your application to use the proxy on port 1080: `127.0.0.1:1080`
+
+### Port Forwarding
+The SOCKS proxy works by port forwarding. Append `.mydcos.directory` to the end of your domain when you enter commands. For instance, `http://example.com:8080/?query=hello` becomes `http://example.com.mydcos.directory:8080/?query=hello`.
+
+##  HTTP
+Run the following command from the DC/OS CLI:
+
+```
+$ sudo dcos tunnel http
+```
+
+To run the HTTP proxy without root privileges, configure it not to use port 80:
+
+```
+$ dcos tunnel http --port 8000
+```
+
+Then, configure your application to run HTTP on the port you specified above.
+
+### Port Forwarding
+The HTTP proxy works by port forwarding. Append `.mydcos.directory` to the end of your domain when you enter commands. For instance, `http://example.com:8080/?query=hello` becomes `http://example.com.mydcos.directory:8080/?query=hello`.
+
+### SRV Records
+The HTTP proxy exposes DC/OS SRV records as URLs in the form `_<port-name>._<task-name>._tcp.marathon.mesos.mydcos.directory`.
+
+To name a port, add `name` to the `portMappings` field of a Marathon application definition:
+
+```json
+"portMappings": [
+        {
+          "containerPort": 3000,
+          "hostPort": 0,
+          "servicePort": 10000,
+          "name": "<my-port-name>",
+          "labels": {
+            "VIP_0": "1.1.1.1:30000"
+          }
+        }
+      ]
+```
+
+Alternatively, [add a named VIP to your application from the DC/OS web interface](https://dcos.io/docs/1.8/usage/service-discovery/load-balancing-vips/virtual-ip-addresses/).
+
+The `<task-name>` is the value of the `id` field in your Marathon application definition or the entry in the **ID** field of a service you create from the DC/OS web interface.
+
+## VPN
+Run the following command from the DC/OS CLI
+
+```
+$ sudo dcos tunnel vpn
+```
+
+The VPN client auto-configures DNS, but this functionality does not work on Mac OSX. To use the VPN client on OSX, [add the following DNS servers](https://support.apple.com/kb/PH18499?locale=en_US):
+
+```
```

Comment:
Oh, this one shouldn't be hard-coded here, the CLI should give them instructions appropriate to their cluster.